locker

```diff
@@ -7,39 +7,58 @@ function lock_image() {
 	echo $NAME@$DIGEST
 }
 
-function lock_dockerfile() {
-	FILE=$1
+function unlock_image() {
+	IMAGE=$1
+	NAME=$(echo $IMAGE | grep '^[^@]\+' -o)
+	echo $NAME
+}
+
+function update_dockerfile() {
+	OPERATION=$1
+	FILE=$2
 	LOCKFILE=$FILE.lock
 
 	while IFS= read -r LINE; do
-		if [[ $LINE =~ ^FROM ]]
+		if [[ $LINE =~ ^FROM ]] && [[ $OPERATION == "" ]]
 		then
 			IMAGE=$(echo "$LINE" | sed -r 's/^FROM +([^ ]+).*$/\1/')
 			LOCKIMAGE=$(lock_image $IMAGE)
 			echo "$LINE" | sed -r "s|$IMAGE|$LOCKIMAGE|" >>$LOCKFILE
+		elif [[ $LINE =~ ^FROM ]] && [[ $OPERATION == "--unlock" ]]
+		then
+			IMAGE=$(echo "$LINE" | sed -r 's/^FROM +([^ ]+).*$/\1/')
+			LOCKIMAGE=$(unlock_image $IMAGE)
+			echo "$LINE" | sed -r "s|$IMAGE|$LOCKIMAGE|" >>$LOCKFILE
 		else
 			echo "$LINE" >>$LOCKFILE
 		fi
 	done <$FILE
 
 	mv $LOCKFILE $FILE
-
-	echo "Locked $FILE"
 }
 
-if [ $# -ne 1 ]
+if [ $# -lt 1 ] || [ $# -gt 2 ]
 then
-	echo "Usage: locker FILE"
+	echo "Usage: locker [-u | --unlock] FILE"
 	exit 1
 fi
 
-FILE=$1
+if [ $# -eq 1 ]
+then
+	FILE=$1
+elif [ $# -eq 2 ]
+then
+	OPTION=$1
+	FILE=$2
+fi
+
```

Comment:
Take a look at `shift` :)

Reply:
I was thinking something like:

* if `$1` is `--unlock` then `shift` and operation is unlock, else operation is lock
* file is `$1`

Shifting allows you to conditionally pop arguments off the stack when they're optional, so you don't have to worry about subsequent positioning.

Reply:
Sure! You lose a bit of safety when it comes to random / mistyped arguments here though, if I'm understanding correctly.

Say I type `locker --unlokc openjdk`, it would actually lock it for me instead? I guess it is fine either way?